cpp/locality/LinkCell.h

```diff
@@ -538,7 +538,19 @@ class LinkCellQueryBallIterator : virtual public LinkCellIterator
                               float r, bool exclude_ii)
         : NeighborQueryIterator(neighbor_query, points, N, exclude_ii),
           LinkCellIterator(neighbor_query, points, N, exclude_ii), m_r(r)
-    {}
+    {
+        // Upon querying, if the search radius is equal to the cell width, we
+        // can guarantee that we don't need to search the cell shell past the
+        // query radius. For simplicity, we store this value as an integer.
+        if (abs(m_r - neighbor_query->getCellWidth()) < 1e-6)
```

Comment:
@vyasr Why not `m_r == neighbor_query->getCellWidth()`? Or at least use a relative tolerance -- I'm hesitant to introduce a scale-dependent parameter like this.

Reply:
That's fair, checking for equality is probably fine. If for whatever reason the simple equality check fails due to some imprecision we're better off falling back to the slower but guaranteed correct version.